scripts/analysis.py

```diff
@@ -0,0 +1,378 @@
+    snf_inventory = cur.execute(exec_string(sink_id,
+                                            'transactions.receiverId',
+                                            'sum(quantity), qualid') + ' group by qualid').fetchall()
+    waste_id = get_waste_id(resources)
+    inven = isotope_calc(waste_id, snf_inventory, cur)
+    return inven
+
+
+def get_sink_agent_ids(cursor):
+    """ Gets all sink agentIds from Agententry table.
+
+        agententry table has the following format:
+            SimId / AgentId / Kind / Spec /
+            Prototype / ParentID / Lifetime / EnterTime
+
+    Parameters
+    ---------
+    cursor: cursor
+        cursor for sqlite3
+
+    Returns
+    -------
+    sink_id: array
+        array of all the sink agentId values.
+    """
+
+    cur = cursor
+    sink_id = []
+    agent = cur.execute("select * from agententry where spec like '%sink%'")
+
+    for ag in agent:
+        sink_id.append(ag[1])
+    return sink_id
+
+
+def get_waste_id(resource_array):
+    """ Gets waste id from a resource array
+
+    Paramters
+    ---------
+    resource_array: array
+        array fetched from the resource table.
+
+    Returns
+    -------
+    waste_id: array
+        array of qualId for waste
+    """
+
+    wasteid = []
+
+    # get all the wasteid
+    for res in resource_array:
+        wasteid.append(res[7])
+
+    return set(wasteid)
+
+
+def exec_string(array, search, whatwant):
+    """ Generates sqlite query command to select things an
+        inner join between resources and transactions.
+
+    Parmaters
+    ---------
+    array: array
+        array of criteria that generates command
+    search: str
+        where [search]
+        criteria for your search
+    whatwant: str
+        select [whatwant]
+        column (set of values) you want out.
+
+    Returns
+    -------
+    exec_str: str
+        sqlite query command.
+    """
+
+    exec_str = ('select ' + whatwant + ' from resources inner join transactions\
+                on transactions.resourceid = resources.resourceid where '
+                + str(search) + ' = ' + str(array[0]))
+
+    for ar in array[1:]:
+        exec_str += ' or ' + str(ar)
+
+    return exec_str
+
+
+def get_sum(array, column_index):
+    """ Returns sum of a column in an array
+
+    Paramters:
+    ---------
+    array: array
+        array that contains a column with numbers
+    column_index: int
+        index for the column to be summed
+    """
+    sum = 0
+    for ar in array:
+        sum += ar[column_index]
+
+    return sum
+
+
+
+def isotope_calc(wasteid_array, snf_inventory, cursor):
+    """ Calculates isotope mass using mass fraction in compositions table.
+
+        Fetches all from compositions table.
+        Compositions table has the following format:
+            SimId / QualId / NucId / MassFrac
+        Then sees if the qualid matches, and if it multiplies
+        the mass fraction by the snf_inventory.
+
+    Prameters
```

Comment:
I never thought you were trolling! No worries. Fun fact, some text editors have a spell check function built-in.